### DISCUSSION TITLE
Allow negative positions in `String.substr` method

### DISCUSSION THREAD

**Discussion by dalexeev** [godotengine/godot-proposals#5481] [Engine Core]
The `left` and `right` methods support negative values.

Current behavior:

```
String substr(from: int, len: int = -1) const

Returns part of the string from the position from with length len. Argument len is optional and using -1
will return remaining characters from given position.
```

I think it makes sense to allow negative `from` values, but I'm not sure about `len`.

**Comment by h0lley** [comment]
negative `from` to go from the end of a string is what I'd expect from a `substr` function.

would negative `len` mean x characters to the left of the `from` position?
in that case the `from` should probably just be offset to the left by that number, and then `len` remain positive.
also it would conflict with the "return remaining characters" feature.

both for negative `from` and `len`, if turned into a proposal, you'd have to reply to "can it be worked around with a few lines of script?" with "yes" though. but I don't think that should immediately exclude it, especially when virtually all substr implementations of comparable languages support negative `from`.
